Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -38,3 +55,224 @@ def test_test_module(requests_mock, mock_client):
+    resp = illumio_virtual_service_create_command(mock_client, {"name": "test_create_virtual_service", "port": "3000"})
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "test_data/create_virtual_service_success_response_protocol_as_udp_hr.md")) as f:
+        expected_hr_output = f.read()
+
+    assert resp.outputs == create_virtual_service_expected_resp
+    assert resp.raw_response == create_virtual_service_expected_resp
+    assert resp.readable_output == expected_hr_output
+    assert resp.outputs_prefix == "Illumio.VirtualService"
+    assert resp.outputs_key_field == "href"
+
+
+def test_illumio_virtual_service_create_command_for_existing_virtual_service(requests_mock, mock_client, capfd):
+    """Test case scenario for illumio-virtual-service-create-command with existing virtual service.
+
+    Given:
+        - illumio_virtual_service_create_command function and mock_client to call the function.
+    When:
+        - Virtual service already exists.
+    Then:
+        - Should raise exception with proper error message.
+    """
+    from IllumioCore import illumio_virtual_service_create_command
+
+    requests_mock.post(VIRTUAL_SERVICE_URL, status_code=406,
+                       json=[{"token": "name_must_be_unique", "message": "Name must be unique"}])
+    with pytest.raises(Exception) as error:
+        capfd.close()
+        illumio_virtual_service_create_command(mock_client,
+                                               {"name": "test_virtual_service", "port": 3000, "protocol": "tcp"})
+
+    assert str(error.value) == "406 Client Error: None for url: {}".format(VIRTUAL_SERVICE_URL)
+
+
+@pytest.mark.parametrize(
+    "err_msg, args",
+    [(NOT_VALID_NUMBER_EXCEPTION_MESSAGE.format("port", "300i0"), {"name": "test", "port": "300i0", "protocol": "tcp"}),
+     (CONVERT_PROTOCOL_EXCEPTION_MESSAGE.format("tcpi"), {"name": "test", "port": "30000", "protocol": "tcpi"}),
+     (MISSING_REQUIRED_PARAM_EXCEPTION_MESSAGE.format("name"), {}),
+     (INVALID_PORT_NUMBER_CREATE_VIRTUAL_SERVICE_EXCEPTION_MESSAGE.format(65536), {"name": "test", "port": "65536"}),
+     (INVALID_PORT_NUMBER_CREATE_VIRTUAL_SERVICE_EXCEPTION_MESSAGE.format(0), {"name": "test", "port": "0"}),
+     (MISSING_REQUIRED_PARAM_EXCEPTION_MESSAGE.format("port"), {"name": "test"})]
+)
+def test_illumio_virtual_service_create_command_when_invalid_arguments_provided(err_msg, args, mock_client, capfd):
+    """Test case scenario for execution of illumio-virtual-service-create-command when invalid argument provided.
+
+    Given:
+        - command arguments for illumio-virtual-service-create-command.
+    When:
+        - Calling `illumio_virtual_service_create_command` function.
+    Then:
+        - Returns a valid error message.
+    """
+    from IllumioCore import illumio_virtual_service_create_command
+
+    with pytest.raises((ValueError, InvalidSingleSelectException)) as err:
+        capfd.close()
```

Comment:
Don't close file descriptors like this, use a generator fixture and close after the yield.

Reply:
okay

Reply:
We don't actually need this, so removing it.